specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,705 @@
+  /// to send the texture to the WebView2 processes to be added to the video stream.
+  /// Do not close or otherwise change the provided `ICoreWebView2Texture` after
+  /// calling this method. Doing so may result in the texture not being added to
+  /// the texture stream and the `ErrorReceived` event may be raised.
+  HRESULT PresentTexture([in] ICoreWebView2StagingTexture* texture)
+
+  /// Stops this texture stream from streaming and moves it into the stopped state.
+  /// When moving to the stopped state the `ICoreWebView2TextureStream Stopped`
+  /// event will be raised and in script the `MediaStreamTrack ended` event will
+  /// be raised. Once stopped, script may again call `Webview.getTextureStream`
+  /// moving the texture stream back to the start requested state.
+  /// See the `StartRequested` event for details.
+
+  /// Once stopped, calls to CreateTexture, GetAvailableTexture,
+  /// PresentTexture, and CloseTexture will fail with
+  /// COREWEBVIEW2_TEXTURE_STREAM_ERROR_NO_VIDEO_TRACK_STARTED.
+  /// See those methods for details.
+
+  /// The `Stop` method is implicitly called when the texture stream object is
+  /// destroyed.
+  HRESULT Stop();
+  /// The `ErrorReceived` event is raised when an error with this texture
+  /// stream occurs asynchronously.
+  HRESULT add_ErrorReceived(
+      [in] ICoreWebView2StagingTextureStreamErrorReceivedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for ErrorReceived event.
+  HRESULT remove_ErrorReceived([in] EventRegistrationToken token);
+  /// Set the D3D device this texture stream should use for creating shared
+  /// texture resources. When the RenderAdapterLUIDChanged event is raised you
+  /// should create a new D3D device using the RenderAdapterLUID property and
+  /// call SetD3DDevice with the new D3D device.
+  /// See the `CreateTextureStream` `d3dDevice` parameter for more details.
+  HRESULT SetD3DDevice([in] IUnknown* d3dDevice);
+  /// Event handler for receiving texture by Javascript.
+  /// `window.chrome.webview.registerTextureStream` call by Javascript will
+  /// request sending video frame to the host where it will filter requested
+  /// page's origin against allowed origins. If allowed, the Javascript will
+  /// send a video frame (web texture), through MediaStreamTrack insertable APIs,
+  /// MediaStreamTrackGenerator.
+  /// https://www.w3.org/TR/mediacapture-transform/.
+  /// WebTextureReceived event will be called only when it receives
+  /// a web texture. There is no start event for receiving web texture.
+  HRESULT add_WebTextureReceived(
+      [in] ICoreWebView2StagingTextureStreamWebTextureReceivedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for WebTextureReceived event.
+  HRESULT remove_WebTextureReceived([in] EventRegistrationToken token);
+  /// Event handler for stopping of the receiving texture stream.

```

Comment:
Do I understand your note "It is expected that the host releases any holding handle/resource from the WebTexture before an event handler returns" that you will close the web textures and the underlying shared memory resources for them after the event handler completes? If so, you need to add a GetDeferral method (see NewWindowRequestedEventArgs.GetDeferral for an example) to the event args. This lets the host app use the deferral to say when the event handler completes asynchronously in case any of the web texture cleanup work that the host needs to do requires running async methods or otherwise requires async work.

```suggestion
  /// The WebTextureStreamStopped event is raised when script unregisters its MediaStream from this texture stream. Script that has previously called `chrome.webview.registerTextureStream`, can call `chrome.webview.unregisterTextureStream` which will raise this event and then close associated ICoreWebView2WebTexture objects after the event handler completes. You should ensure that you release any references to associated ICoreWebView2WebTexture objects and their underlying resources before the event handler completes.
  /// Once stopped, script may start again by calling `chrome.webview.registerTextureStream` and sending more frames. In this case the `ICoreWebView2TextureStream WebTextureReceived` event will be raised again.
```

Reply:
WetTextureStopped event is a notification that the browser already released all shared resources. It is expected that the host cleans any resources the host created for supporting the received web texture.

slightly rephrase the first one.
"/// The WebTextureStreamStopped event is raised when script unregisters its
  /// MediaStream from this texture stream. Script that has previously called
  /// `chrome.webview.registerTextureStream`, can call `chrome.webview.
  /// unregisterTextureStream` which will raise this event and then close
  /// associated ICoreWebView2WebTexture objects in the browser side. You should
  /// ensure that you release any references to associated
  /// ICoreWebView2WebTexture objects and their underlying resources."